File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Reports/Commands/ReportCommand.cs
using SIE.Web.Command;
using System;

namespace SIE.Web.MES.TaskManagement.Reports.Commands
{
    /// <summary>
    /// 报工
    /// </summary>
    [JsCommand("SIE.Web.MES.TaskManagement.Reports.ReportCommand")]
    public class ReportCommand : ViewCommand
    {
        public const string FullName = "SIE.Web.MES.TaskManagement.Reports.ReportCommand";

        protected override object Excute(ViewArgs args, string scope)
        {
            throw new NotImplementedException();
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Reports/Commands/ReportCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Reports.ReportCommand', {
    extend: 'SIE.cmd.Command',
    meta: { text: "报工", group: "edit", iconCls: "icon-ArrowWithCircleDown icon-green" },
    executeIntervalMode: SIE.cmd.IntervalMode.Debounce.value,
    canExecute: function (view) {        
        return view._parent.getSelection().length == 1 && view._parent._current.data.TaskStatus == 30 && view._parent._current.data.ReportMode == 1;
    },
    /*
   * @override 执行
   * @returns{}
   */
    execute: function (view, source) {
        var me = this;
        SIE.Msg.wait("正在报工......".t());
        SIE.Web.MES.TaskManagement.Reports.ReportCommon.saveOrSubmmitReport(view, true, me.meta.command);
    },
});